face.go

```diff
@@ -1,3 +1,102 @@
+// Package font provides supports for parsing
+// several font formats (postscript, bitmap and truetype)
+// and provides a common API.
+//
+// It provides low-level metric information used by shaping
+// engines and a way of rendering outline and bitmap glyphs.
 package font
 
-type Face interface{}
+// Face is the common interface to represent various font formats.
+//
+// A Face object is the in-memory representation of a font file:
+// as such it does not hold scaling information.
+//
+// Complex font features may require an extension interface :
+// see opentype.FaceOpentype for an example.
+type Face interface {
+	FaceMetrics
+	FaceRenderer
+}
+
+// FaceMetrics exposes the information about a font file
+// required by a layout engine.
+// All the returned metrics are expressed in font units
+// and should be scaled.
+type FaceMetrics interface {
+	// Upem returns the units per em of the font file.
+	// If not found, should return 1000 as fallback value.
+	Upem() uint16
+
+	// GlyphName returns the name of the given glyph, or an empty
+	// string if the glyph is invalid or has no name.
+	GlyphName(glyph GID) string
+
+	// FontHExtents returns the extents of the font for horizontal text, or false
+	// it not available, in font units.
+	// `varCoords` (in normalized coordinates) is only useful for variable fonts.
+	FontHExtents(varCoords []float32) (FontExtents, bool)
```

Comment:
Yes, you're right. 

Variable fonts may be seen as parametrized fonts, where the user will specify (at run time) the values for these parameters. More precisely, the parameters are scalars between -1 and 1 (once normalized), and the number of parameters depends on the font (typical values would be between 1 and 5). For example, a variable font may expose a Weight parameters, where -1 would correspond to a very thin font, 0 to a normal font and 1 to a thick bold font, with all the nuances possible between the two.

From an API point of view, this means that a variable font requires, in addition to a classical font file, the values of the parameters, hence the presence of `varCoords` in several methods.

Thinking of it, an alternative design could move this settings to a constructor of the implementations of `FaceMetrics`. I think it would make sense since typically these parameters are fixed once for all. Then the methods of `FaceMetrics` interface would be simpler.
The drawback I see is that we would loose the property that a `FaceMetrics` is a direct in memory representation of a font file (without user specified information). 
On the other hand, from what I know, a variable font, once set up (with the `varCoords` parameters), behaves just like any other font (the parameters only acting internally).

Reply:
with the caveat that I am clearly an ignorant about fonts, I believe the API w/o `varCoords` looks "cleaner".
and if that can be handled by a different constructor, hiding that "detail", all the better.